hw/xbox/dsp/dsp_cpu.c

```diff
@@ -420,23 +429,30 @@ void dsp56k_reset_cpu(dsp_core_t* dsp)
     dsp->disasm_prev_inst_pc = 0xFFFFFFFF;
 }
 
-static OpcodeEntry lookup_opcode(uint32_t op) {
-    OpcodeEntry r = {0};
+static OpcodeEntry* search_opcode(uint32_t op)
+{
+    OpcodeEntry* r;
     int i;
     for (i=0; i<ARRAYSIZE(nonparallel_opcodes); i++) {
         if ((op & nonparallel_matches[i][0]) == nonparallel_matches[i][1]) {
             if (nonparallel_opcodes[i].match_func 
                 && !nonparallel_opcodes[i].match_func(op)) continue;
-            if (r.template != NULL) {
-                printf("qqq %x %s\n", op, r.template);
+            if (r->template != NULL) {
```

Comment:
Did you try running this code?

`r` is not initialized at this point (should be `r = NULL`).

If you change from the object, which was `{0}` initialized, to pointers, then you have to do a different check here: `(r != NULL) && (r->template != NULL)` to make sure that no other opcode was found yet (which seems to be the idea behind this print), and to make sure it has a template.

You , as you have `assert(r->template == NULL)`